chapters/en/chapter1/7.mdx

```diff
@@ -1,270 +1,257 @@
-	src="https://agents-course-beam-search-visualizer.hf.space"
-	frameborder="0"
-	width="850"
-	height="450"
-></iframe>
-
-This approach often produces more coherent and grammatically correct text, though it requires more computational resources than simpler methods.
+So far, this chapter has covered a lot of ground! Don't worry if you didn't grasp all the details, but it's to reflect on what you've learned so far with a quiz. 
+
+This quiz is ungraded, so you can try it as many times as you want. If you struggle with some questions, follow the tips and revisit the material. You'll be quizzed on this material again in the certification exam.
+
+### 1. Explore the Hub and look for the `roberta-large-mnli` checkpoint. What task does it perform?
+
+
+<Question
+	choices={[
+		{
+			text: "Summarization",
+			explain: "Look again on the <a href=\"https://huggingface.co/roberta-large-mnli\">roberta-large-mnli page</a>."
+		},
+		{
+			text: "Text classification",
+			explain: "More precisely, it classifies if two sentences are logically linked across three labels (contradiction, neutral, entailment) — a task also called <em>natural language inference</em>.",
+			correct: true
+		},
+		{
+			text: "Text generation",
+			explain: "Look again on the <a href=\"https://huggingface.co/roberta-large-mnli\">roberta-large-mnli page</a>."
+		}
+	]}
+/>
 
-## Practical Challenges and Optimization
+### 2. What will the following code return?
+
+```py
+from transformers import pipeline
+
+ner = pipeline("ner", grouped_entities=True)
+ner("My name is Sylvain and I work at Hugging Face in Brooklyn.")
+```
+
+<Question
+	choices={[
+		{
+			text: "It will return classification scores for this sentence, with labels \"positive\" or \"negative\".",
+			explain: "This is incorrect — this would be a <code>sentiment-analysis</code> pipeline."
+		},
+		{
+			text: "It will return a generated text completing this sentence.",
+			explain: "This is incorrect — it would be a <code>text-generation</code> pipeline.",
+		},
+		{
+			text: "It will return the words representing persons, organizations or locations.",
+			explain: "Furthermore, with <code>grouped_entities=True</code>, it will group together the words belonging to the same entity, like \"Hugging Face\".",
+			correct: true
+		}
+	]}
+/>
 
-As we wrap up our exploration of LLM inference, let's look at the practical challenges you'll face when deploying these models, and how to measure and optimize their performance.
+### 3. What should replace ... in this code sample?
+
+```py
+from transformers import pipeline
+
+filler = pipeline("fill-mask", model="bert-base-cased")
+result = filler("...")
+```
+
+<Question
+	choices={[
+		{
+			text: "This &#60;mask> has been waiting for you.",
+			explain: "This is incorrect. Check out the <code>bert-base-cased</code> model card and try to spot your mistake."
+		},
+		{
+			text: "This [MASK] has been waiting for you.",
+			explain: "Correct! This model's mask token is [MASK].",
```

Comment:
```suggestion
			explain: "This model's mask token is [MASK].",
```
Since the answer is correct, it adds `Correct!` so removing one not to duplicate `Correct! Correct! This model's mask token is [MASK].`